### PR TITLE
Add daml start

### PR DESCRIPTION
**Pull request by moritzkiefer-da**
This adds a first version of a `daml start` command that starts the sandbox and the navigator on a given DAR file. It is not quite the same as `da start` which also compiles the DAR automatically but we can change it later. The main logic added in this PR is the one for starting the sandbox and navigator and that will be necessary regardless of what exactly we end up choosing as the behavior for `daml start` (and we also need that logic for integration tests).

I also shuffled a bunch of code to `daml-project-config`, in particular `DAML.Assistant.Config` and the remainder of `DAML.Assistant.Consts` as well as the dependencies of those modules. I haven’t changed the code there.

### CODE REVIEW

File: daml-assistant/daml-helper/DamlHelper.hs [hunk 1]
```diff
@@ -103,6 +132,58 @@ runListTemplates = do
...
+            let navigatorConfPath = confDir </> "navigator-config.json"
+            writeFileUTF8 navigatorConfPath (T.unpack $ navigatorConfig parties)
+            withNavigator sandboxPort navigatorPort navigatorConfPath [] $ \navigatorPh ->
+                void $ race (waitForProcess navigatorPh) (waitForProcess sandboxPh)
+    where sandboxPort = SandboxPort 6865
+          navigatorPort = NavigatorPort 7500
+
+getProjectParties :: IO [T.Text]
+getProjectParties = do
+    projectPath <- required "Must be called from within a project" =<< getProjectPath
```
Comment (martin-drhu-da) [inline daml-assistant/daml-helper/DamlHelper.hs]: This looks like something that might be useful in multiple places. Should we move it into its own function?

File: daml-assistant/daml-helper/DamlHelper.hs [hunk 2]
```diff
@@ -121,3 +202,39 @@ installExtension src target =
...
+
+-- | `waitForConnectionOnPort sleep port` keeps trying to establish a TCP connection on the given port.
+-- Between each connection request it calls `sleep`.
+waitForConnectionOnPort :: IO () -> Int -> IO ()
+waitForConnectionOnPort sleep port = do
+    let hints = defaultHints { addrFlags = [AI_NUMERICHOST, AI_NUMERICSERV], addrSocketType = Stream }
+    addr : _ <- getAddrInfo (Just hints) (Just "127.0.0.1") (Just $ show port)
+    untilJust $ do
+        r <- tryIO $ checkConnection addr
+        either (const $ Nothing <$ sleep) (const $ pure $ Just ()) r
```
Comment (martin-drhu-da) [inline daml-assistant/daml-helper/DamlHelper.hs]: This is almost unintelligible. Can we please use a `case` expression. :D

Comment (moritzkiefer-da) [reply daml-assistant/daml-helper/DamlHelper.hs]: Changed

File: daml-assistant/daml-helper/DamlHelper.hs (hunk 1 above)
Comment (moritzkiefer-da) [reply daml-assistant/daml-helper/DamlHelper.hs]: I’ll do so when we use it somewhere else.

**Pull request merged**